src/ruby/puppet-server-lib/puppet/server/certificate.rb

```diff
@@ -15,6 +16,7 @@ def initialize(java_cert)
     end
 
     @java_cert = java_cert
+    @oid_defns = Puppet::Server::Oids.load_custom_oid_file(Puppet[:trusted_oid_mapping_file]) + Puppet::SSL::Oids::PUPPET_OIDS
```

Comment:
The difference I see here vs. the csr attributes is that that code already lived in core Puppet.  Here we're introducing net new parsing logic in the Puppet Server repo so I feel like we have a lot more control and flexibility over it.

I'm pretty strongly opposed to have the Clojure services reach into the Ruby world to get the data.

I'm not seeing a reason why we couldn't parse the file a single time, in Clojure, at startup, and then just inject the required data into the Ruby containers?